src/yarp.c

```diff
@@ -2562,6 +2562,9 @@ lex_token_type(yp_parser_t *parser) {
           if (lex_state_operator_p(parser)) {
             lex_state_set(parser, YP_LEX_STATE_ARG);
           } else {
+            if (lex_state_p(parser, YP_LEX_STATE_ARG)) {
```

Comment:
I don't think we should do this in the lexer. I think you're trying to mirror the behavior from the CRuby parser here: https://github.com/ruby/ruby/blob/d579f47558eaa27a989c94996bf71b6cc1047ff2/parse.y#L4019-L4023. In that case we should perform this state change in the parser not the lexer.